testdata/p4tc_samples/default_action_with_param_01.p4

```diff
@@ -0,0 +1,135 @@
+#include <core.p4>
+#include <tc/pna.p4>
+
+typedef bit<48>  EthernetAddress;
+
+header ethernet_t {
+    EthernetAddress dstAddr;
+    EthernetAddress srcAddr;
+    bit<16>         etherType;
+}
+
+header ipv4_t {
+    bit<4>  version;
+    bit<4>  ihl;
+    bit<8>  diffserv;
+    bit<16> totalLen;
+    bit<16> identification;
+    bit<3>  flags;
+    bit<13> fragOffset;
+    bit<8>  ttl;
+    bit<8>  protocol;
+    bit<16> hdrChecksum;
+    @tc_type ("ipv4") bit<32> srcAddr;
+    @tc_type ("ipv4") bit<32> dstAddr;
+}
+
+//////////////////////////////////////////////////////////////////////
+// Struct types for holding user-defined collections of headers and
+// metadata in the P4 developer's program.
+//
+// Note: The names of these struct types are completely up to the P4
+// developer, as are their member fields, with the only restriction
+// being that the structs intended to contain headers should only
+// contain members whose types are header, header stack, or
+// header_union.
+//////////////////////////////////////////////////////////////////////
+
+struct main_metadata_t {
+    // empty for this skeleton
+}
+
+// User-defined struct containing all of those headers parsed in the
+// main parser.
+struct headers_t {
+    ethernet_t ethernet;
+    ipv4_t     ipv4;
+}
+
+parser MainParserImpl(
+    packet_in pkt,
+    out   headers_t hdr,
+    inout main_metadata_t main_meta,
+    in    pna_main_parser_input_metadata_t istd)
+{
+    state start {
+        pkt.extract(hdr.ethernet);
+        transition select(hdr.ethernet.etherType) {
+            0x0800  : parse_ipv4;
+            default : accept;
+        }
+    }
+    state parse_ipv4 {
+        pkt.extract(hdr.ipv4);
+        transition accept;
+    }
+}
+
+control MainControlImpl(
+    inout headers_t hdr,                 // from main parser
+    inout main_metadata_t user_meta,     // from main parser, to "next block"
+    in    pna_main_input_metadata_t istd,
+    inout pna_main_output_metadata_t ostd)
+{
+    PortId_t z = (PortId_t)4;
+    action next_hop(inout PortId_t vport) {
+        send_to_port(vport);
+    }
+    action default_route_drop() {
```

Comment:
Another small nit, could you change this function's name to `dflt_route_drop`?
P4TC has a limit on the action name length of 32 bytes and we concatenate the control block name with the action name when sending to the kernel. So the action name for the kernel is `MainControlImpl/default_route_drop` which has 34 bytes.
It would also be interesting if the compiler rejected actions with names that are too large, but this is unrelated to this PR

Reply:
I will change this case.
>It would also be interesting if the compiler rejected actions with names that are too large, but this is unrelated to this PR

Should the compiler issue error in this case?
If yes, could you please create new issue in https://github.com/p4tc-dev/p4tc-e2e/issues. So we can track it

Reply:
> I will change this case.
> 
> > It would also be interesting if the compiler rejected actions with names that are too large, but this is unrelated to this PR
> 
> Should the compiler issue error in this case?

Yes

> If yes, could you please create new issue in https://github.com/p4tc-dev/p4tc-e2e/issues. So we can track it

Done (https://github.com/p4tc-dev/p4tc-e2e/issues/59)

Reply:
Testcase updated. Please confirm once testing is completed from your side. Will raise opensource PR